src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs

```diff
@@ -415,6 +415,19 @@ public static ImmutableArray<ITypeParameterSymbol> GetTypeParameters(this ISymbo
             }
         }
 
+        public static ImmutableArray<ITypeParameterSymbol> GetAllTypeParameters(this ISymbol symbol)
```

Comment:
you get these in a very weird order.  They go from innermost symbol to outermost.  Is that desirable?